### PR TITLE
Redeem request handler

### LINKED ISSUE #144: Redeem request handler
- stakeRequests repository will be renamed to requestRepository table. Both stake and redeem schema is similar.

- A new field requestType will be added to requestRepository. Values will be stake/redeem.

### LINKED ISSUE #149: Redeem request model

### LINKED ISSUE #150: Redeem request repository

### PR DESCRIPTION
**Pull request by abhayks1**
PR covers below:

- Implementation of Redeem request handler
- Renaming of stakeRequestRepository to requestRepository to support  redeemRequests.
- Renaming of stakeRequest model to Request model.
- New column requestType has been added in requestRepository. Values are stake/redeem.

Fixes #144 
Fixes #149 
Fixes #150

### CODE REVIEW

File: src/models/Request.ts [hunk 1]
```diff
@@ -48,43 +50,45 @@ export default class StakeRequest extends Comparable<StakeRequest> {
   public updatedAt?: Date;
 
   public constructor(
-    stakeRequestHash: string,
+    requestHash: string,
+    requestType: string,
     blockNumber: BigNumber,
     amount?: BigNumber,
     beneficiary?: string,
```
Comment (sarvesh-ost) [inline src/models/Request.ts]: why these parameters are optional?

File: src/repositories/RequestRepository.ts [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+          allowNull: false,
+          validate: {
+            min: 0,
+          },
+        },
+        beneficiary: {
+          type: DataTypes.STRING,
+          allowNull: false,
+        },
+        gasPrice: {
```
Comment (sarvesh-ost) [inline src/repositories/RequestRepository.ts]: Model has optional argument for this field but here it's required.

File: test/handlers/redeem_and_unstake/RedeemRequestedHandler/persist.test.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      models.length,
+      transactions.length,
+      'Number of models must be equal to transactions',
+    );
+    assert.deepStrictEqual(models[0], redeemRequest);
+    SpyAssert.assert(saveStub, 1, [[redeemRequest]]);
+    sinon.restore();
+  });
+
+  it('should update messageHash(null) and blockNumber when redeemRequest '
```
Comment (sarvesh-ost) [inline test/handlers/redeem_and_unstake/RedeemRequestedHandler/persist.test.ts]:
```suggestion
  it('should update blockNumber and messageHash with blank  when redeemRequest '
```

File: src/models/Request.ts [hunk 4]
```diff
@@ -20,8 +20,10 @@ import BigNumber from 'bignumber.js';
 
 import Comparable from '../observer/Comparable';
 
-export default class StakeRequest extends Comparable<StakeRequest> {
-  public stakeRequestHash: string;
+export default class Request extends Comparable<Request> {
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/models/Request.ts]: Done!

File: test/handlers/redeem_and_unstake/RedeemRequestedHandler/persist.test.ts (hunk 3 above)
Comment (abhayks1) [reply test/handlers/redeem_and_unstake/RedeemRequestedHandler/persist.test.ts]: Done!

File: src/models/Request.ts (hunk 1 above)
Comment (abhayks1) [reply src/models/Request.ts]: There are cases where stakeRequest object is created with mandatory arguments. It was existing logic. Also in multiple tests, stakeRequest is created with mandatory arguments.
e.g. method: updateMessageHashInRequestRepository in AcceptStakerequestService.

File: src/models/Request.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (abhayks1) [reply src/models/Request.ts]: We had discussion around this and we agreed to keep Request model name which can handle both stake/redeem requests in Facilitator context.

File: src/repositories/RequestRepository.ts (hunk 2 above)
Comment (abhayks1) [reply src/repositories/RequestRepository.ts]: As discussed created below ticket to handle this:
https://github.com/mosaicdao/facilitator/issues/188

File: src/models/MessageTransferRequest.ts [hunk 5]
```diff
@@ -20,8 +20,13 @@ import BigNumber from 'bignumber.js';
 
 import Comparable from '../observer/Comparable';
 
-export default class StakeRequest extends Comparable<StakeRequest> {
-  public stakeRequestHash: string;
+/**
+ * It Represents MessageTransferRequest model object. It stores stake and redeem requests.
```
Comment (sarvesh-ost) [inline src/models/MessageTransferRequest.ts]:
```suggestion
 * It represents MessageTransferRequest model object. It stores stake and redeem requests.
```

**Pull request merged**